packages/gatsby/src/schema/resolvers.ts

```diff
@@ -566,13 +566,23 @@ export function wrappingResolver<TSource, TArgs>(
       )
       activity.start()
     }
-    try {
-      return resolver(parent, args, context, info)
-    } finally {
+    const result = resolver(parent, args, context, info)
+
+    if (!activity) {
+      return result
+    }
+
+    const endActivity = (): void => {
       if (activity) {
```

Comment:
This can be dropped since we test and shortcut return above

Reply:
Typescript disagrees.

This file has incomplete typings but if I add proper typings to `activity` and remove this check I get TS error: `TS2339: Property 'end' does not exist on type 'void | IActivity'.   Property 'end' does not exist on type 'void'.`